tests/integration/test_bundler.py

```diff
@@ -53,6 +53,34 @@
             ),
             id="bundler_malformed_lockfile",
         ),
+        pytest.param(
+            utils.TestParameters(
+                repo="https://github.com/cachito-testing/cachi2-bundler.git",
+                ref="well_formed_ruby_all_features",
+                packages=({"path": ".", "type": "bundler"},),
+                flags=["--dev-package-managers"],
+                check_output=False,
+                check_deps_checksums=False,
+                check_vendor_checksums=False,
+                expected_exit_code=0,
+                expected_output="",
+            ),
+            id="bundler_everything_present",
+        ),
+        pytest.param(
+            utils.TestParameters(
+                repo="https://github.com/cachito-testing/cachi2-bundler.git",
+                ref="well_formed_ruby_without_gemspec",
+                packages=({"path": ".", "type": "bundler"},),
+                flags=["--dev-package-managers"],
+                check_output=False,
+                check_deps_checksums=False,
+                check_vendor_checksums=False,
+                expected_exit_code=0,
+                expected_output="",
+            ),
+            id="bundler_everything_present_except_gemspec",
+        ),
```

Comment:
Is it worth adding a scenario with multiple packages in the same repo ?

Reply:
Probably not, at least not with our current ITs structure. This scenario would essentially exercise the same code that is exercised in existing tests, but in a loop.

Reply:
One last thing. We don't have any e2e tests and I believe we could make one of those tests above (that are almost identical) to do full pre-fetch and build an image.

Reply:
Good point, I'll add a e2e test in a separate commit to this PR (after it passes locally).